src/cli.c

```diff
@@ -0,0 +1,115 @@
+#include "cli.h"
+#include "logging.h"
+
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <limits.h>
+#include <getopt.h>
+#include <stdio.h>
+#include <errno.h>
+
+/******************************************************
+ *** LOCAL DEFINES                                  ***
+ ******************************************************/
```

Comment:
What exactly do you mean here? 
I personally really like this sectioning, as this provides a clear structure and ensures that everything is in one place.

Reply:
I'm sorry I probably didn't elaborate enough. I was referring to granularity, e.g., a variables section instead of one for local and global.
I think that sorting defines, variables and functions is a good thing, also local before global, but I'm not sure each type needs a section. When you have a file with a single function and a single variable it gets really cluttered 🤔

Reply:
I would argue that having sections for local/exposed methods makes sense, but I would also omit these distinctions for defines/variables.
Additionally, as mentioned below, I would also omit empty sections as this clutters the files unnecessarily and can always be re-added if necessary

Reply:
Well that makes it 2:1, so I'll remove these.

Reply:
But personally I don't think that this clutters the file. Even if sections are empty, you can for example see, that there are not file global variables to worry about. If empty sections are omitted, nobody is going to add new sections should they be necessary. I suppose its either no sectioning at all and just making sure that the functions are ordered accordingly, or all sections, be they empty or not.

Reply:
Why would you omit the distinction between local variables and defines @Benzammour?

Reply:
Actually, I meant the distinction between local/exposed variables and local/exposed `defines`.
Exposed `defines` are in the appropriate `.h` file anyway, and for variables, it's eminent either via ordering or looking into the `.h` file. If you only use `LOCAL *` as a prefix, then ignore the comments about the defines/variables

Regarding the sections, however, I would still argue against them.   
In that case, we still know that there is no file-global variable to worry about by the absence of its header.  
And if someone does not add new sections where necessary, this change will not be merged until it is corrected due to not adhering to our code guidelines.

Reply:
That's fair. For defines specifically, LOCAL would be the only prefix. 
I removed the sectioning completely as of now. I gather that you would keep it but only for the sections that are needed?

Reply:
Exactly

Reply:
I'll re-add them then.